cmd/cloud-controller-manager/app/controllermanager.go

```diff
@@ -273,11 +316,45 @@ var ControllersDisabledByDefault = sets.NewString()
 
 // newControllerInitializers is a private map of named controller groups (you can start more than one in an init func)
 // paired to their initFunc.  This allows for structured downstream composition and subdivision.
-func newControllerInitializers() map[string]initFunc {
+func newControllerInitializers(controllerConfig controllers.Config) map[string]initFunc {
 	controllers := map[string]initFunc{}
-	controllers["cloud-node"] = startCloudNodeController
 	controllers["cloud-node-lifecycle"] = startCloudNodeLifecycleController
-	controllers["service"] = startServiceController
-	controllers["route"] = startRouteController
+
+	_, withMigration := contrllerConfig.ControllerConfig("service-controller")
```

Comment:
A little worried we may be too tightly proscribing which controllers can be migrated. It seems like we should enable *all* controllers to be migratable. 

It might also be nice if we had some sort of structure for the controllers. Something like name, start method, default component/CM. It might also (later) indicate things like which resources it needs.

Reply:
Agreed, we can probably abstract this better in struct or something if we go forward with implementatoin